legate/core/transform.py

```diff
@@ -382,62 +450,61 @@ def convert(self, partition):
                 f"Unsupported partition: {type(partition).__name__}"
             )
 
-    def convert_restrictions(self, restrictions):
+    def convert_restrictions(self, restrictions: Restrictions) -> Restrictions:
         return tuple(restrictions[idx] for idx in self._axes)
 
-    def get_inverse_transform(self, ndim):
+    def get_inverse_transform(self, ndim: int) -> AffineTransform:
         result = AffineTransform(ndim, ndim, False)
         for dim in range(ndim):
             result.trans[self._axes[dim], dim] = 1
-        return result, ndim
+        return result
 
-    def serialize(self, buf):
-        super(Transpose, self).serialize(buf)
+    def serialize(self, buf: BufferBuilder) -> None:
+        code = self._runtime.get_transform_code(self.__class__.__name__)
+        buf.pack_32bit_int(code)
         buf.pack_32bit_uint(len(self._axes))
         for axis in self._axes:
             buf.pack_32bit_int(axis)
 
 
 class Delinearize(Transform):
-    def __init__(self, runtime, dim, shape):
+    def __init__(self, runtime: Runtime, dim: int, shape: Shape) -> None:
         self._runtime = runtime
         self._dim = dim
         self._shape = Shape(shape)
         self._strides = self._shape.strides()
 
-    def compute_shape(self, shape):
+    def compute_shape(self, shape: Shape) -> Shape:
         return shape.replace(self._dim, self._shape)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"Delinearize(dim: {self._dim}, shape: {self._shape})"
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         if not isinstance(other, type(self)):
             return False
         return self._dim == other._dim and self._shape == other._shape
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash((type(self), self._shape, self._strides))
 
-    def adds_fake_dims(self):
+    def adds_fake_dims(self) -> bool:
         return False
 
     @property
-    def convertible(self):
+    def convertible(self) -> bool:
         return False
 
-    def invert(self, partition):
+    def invert(self, partition: PartitionBase) -> PartitionBase:
         if isinstance(partition, Tiling):
-            if (
-                partition.color_shape[
-                    self._dim + 1 : self._dim + self._shape.ndim
-                ].volume()
-                == 1
-                and partition.offset[
-                    self._dim + 1 : self._dim + self._shape.ndim
-                ].sum()
-                == 0
-            ):
+            color_shape = partition.color_shape[
+                self._dim + 1 : self._dim + self._shape.ndim
+            ]
+            offset = partition.offset[
+                self._dim + 1 : self._dim + self._shape.ndim
+            ]
+
+            if color_shape.volume() == 1 and offset.sum() == 0:
```

Comment:
In general I will try to optimize to make conditions short/readable on their own (esp with  `black` formatting and a short line limit)